node/external/blockAPI/metaBlock.go

```diff
@@ -170,41 +172,109 @@ func (mbp *metaAPIBlockProcessor) getHashAndBlockBytesFromStorerByNonce(params a
 }
 
 func (mbp *metaAPIBlockProcessor) convertMetaBlockBytesToAPIBlock(hash []byte, blockBytes []byte, options api.BlockQueryOptions) (*api.Block, error) {
-	blockHeader := &block.MetaBlock{}
-	err := mbp.marshalizer.Unmarshal(blockHeader, blockBytes)
+	blockHeader, err := process.UnmarshalMetaHeader(mbp.marshalizer, blockBytes)
 	if err != nil {
 		return nil, err
 	}
 
+	notarizedBlocks := make([]*api.NotarizedBlock, 0, len(blockHeader.GetShardInfoHandlers()))
+	for _, shardData := range blockHeader.GetShardInfoHandlers() {
+		notarizedBlock := &api.NotarizedBlock{
+			Hash:  hex.EncodeToString(shardData.GetHeaderHash()),
+			Nonce: shardData.GetNonce(),
+			Round: shardData.GetRound(),
+			Shard: shardData.GetShardID(),
+		}
+
+		notarizedBlocks = append(notarizedBlocks, notarizedBlock)
+	}
+
+	apiMetaBlock := &api.Block{
+		Nonce:           blockHeader.GetNonce(),
+		Round:           blockHeader.GetRound(),
+		Epoch:           blockHeader.GetEpoch(),
+		Shard:           core.MetachainShardId,
+		Hash:            hex.EncodeToString(hash),
+		PrevBlockHash:   hex.EncodeToString(blockHeader.GetPrevHash()),
+		NotarizedBlocks: notarizedBlocks,
+		Timestamp:       int64(blockHeader.GetTimeStamp()),
+		TimestampMs:     int64(common.ConvertTimeStampSecToMs(blockHeader.GetTimeStamp())),
+		Status:          BlockStatusOnChain,
+		PubKeyBitmap:    hex.EncodeToString(blockHeader.GetPubKeysBitmap()),
+		Signature:       hex.EncodeToString(blockHeader.GetSignature()),
+		LeaderSignature: hex.EncodeToString(blockHeader.GetLeaderSignature()),
+		ChainID:         string(blockHeader.GetChainID()),
+		SoftwareVersion: hex.EncodeToString(blockHeader.GetSoftwareVersion()),
+		ReceiptsHash:    hex.EncodeToString(blockHeader.GetReceiptsHash()),
+		Reserved:        blockHeader.GetReserved(),
+		RandSeed:        hex.EncodeToString(blockHeader.GetRandSeed()),
+		PrevRandSeed:    hex.EncodeToString(blockHeader.GetPrevRandSeed()),
+	}
+
+	if !blockHeader.IsHeaderV3() {
```

Comment:
should we correlate this in any way with the supernova activation flag? the first header v3 will be when async execution will be fully enabled?